centreon/www/front_src/src/Dashboards/useUserDashboardPermissions.ts

```diff
@@ -0,0 +1,33 @@
+import { useAtomValue } from 'jotai';
+import { equals } from 'ramda';
+
+import { userAtom } from '@centreon/ui-context';
+
+import { DashboardRole, Dashboard } from './models';
+
+interface useUserDashboardPermissionsState {
+  getHasEditPermission: (dashboard: Dashboard) => boolean;
+  isEitherCreatorOrAdministrator: boolean;
+}
+
+const useUserDashboardPermissions = (): useUserDashboardPermissionsState => {
+  const { dashboard: globalRoles } = useAtomValue(userAtom);
+
+  const isEitherCreatorOrAdministrator =
+    globalRoles.createRole || globalRoles.administrateRole;
+
+  const getHasEditPermission = (dashboard: Dashboard): boolean => {
+    return (
+      globalRoles.administrateRole ||
+      (isEitherCreatorOrAdministrator &&
+        equals(dashboard.ownRole, DashboardRole.editor))
+    );
+  };

```

Comment:
imo. we should not be translating the roles to permissions in the front, but use the permissions returned from the back-end (single source of truth)


```
  const canEditPermission = (dashboard: Dashboard): boolean => {
    return (
      globalPermissions.administrateDashboards ||
      (globalPermissions.editDashboards &&
        equals(dashboard.ownRole, DashboardRole.editor))
    );
  };

```

Reply:
How would you do that without duplicating the code?